torch/distributions/distribution.py

```diff
@@ -241,6 +241,9 @@ def _extended_shape(self, sample_shape=torch.Size()):
         Args:
             sample_shape (torch.Size): the size of the sample to be drawn.
         """
+        if isinstance(sample_shape, int):
```

Comment:
How about making this more general?

```
from collections.abc import Iterable

if not isinstance(sample_shape, Iterable):
    raise TypeError(f"Expected `sample_shape` to be an iterable of int, found {type(sample_shape)}."
```

@fritzo: Instead of raising an error for int types, is there an issue with promoting this to `torch.Size([sample_shape])` here? I think this was likely discussed earlier at some point and you had good reasons for not doing this, but I don't recall at the moment.

Reply:
promoting to `torch.Size` sounds fine to me.  I don't recall the previous error.  I guess we should check that tracing / compiling still works.  I'm not sure we have enough tests

Reply:
@neerajprad @fritzo I was checking how `torch.zeros(1)` and `torch.zeros((1,))` or `torch.zeros(1,2)` and `torch.zeros((1,2))` produce the same output. I found that [this method](https://github.com/pytorch/pytorch/blob/7070fe4d1596ee1b7d8b42b551acb497ded10867/torch/distributed/_shard/sharded_tensor/utils.py#L94) is used by `torch.zeros`, `torch.rand` and maybe other methods to adapt the user input to desired `torch.Size`. How about directly importing it and using it in this case?